src/altinn-app-frontend/src/components/base/DatepickerComponent.test.tsx

```diff
@@ -97,133 +97,81 @@ describe('DatepickerComponent', () => {
     );
   });
 
-  it('should call handleDataChange with correct value when timeStamp is undefined when field is changed with a valid date', async () => {
+  it('should call handleDataChange without skipping validation if date is cleared', async () => {
     const handleDataChange = jest.fn();
-    render({ handleDataChange, timeStamp: undefined });
+    render({ handleDataChange, formData: { simpleBinding: '2022-01-01' } });
 
     const inputField = screen.getByRole('textbox');
 
-    await userEvent.type(inputField, '12.26.2022');
+    await userEvent.clear(inputField);
+    fireEvent.blur(inputField);
 
-    expect(handleDataChange).toHaveBeenCalledWith(
-      // Ignore TZ part of timestamp to avoid test failing when this changes
-      expect.stringContaining('2022-12-26T12:00:00.000+'),
-    );
+    expect(handleDataChange).toHaveBeenCalledTimes(1);
+    expect(handleDataChange).toHaveBeenCalledWith('');
   });
 
-  it('should call handleDataChange with correct value when timeStamp is true when field is changed with a valid date', async () => {
+  it('should call handleDataChange with formatted value (timestamp=true) without skipping validation if date is valid', async () => {
     const handleDataChange = jest.fn();
     render({ handleDataChange, timeStamp: true });
 
     const inputField = screen.getByRole('textbox');
 
-    await userEvent.type(inputField, '12.26.2022');
+    await userEvent.type(inputField, '01012022');
```

Comment:
Maybe use the 31st of December as test data instead? That will catch if something goes wrong at some point switching the date/month. When the date and month is the same, these things might not get caught.

Reply:
Yeah, I think the reason I didn't was that it looks that the format used comes from moment.locale, and this may not be 100% predictable? I guess i can make sure that the format is specified explicitly, and then use a day > 12

Reply:
I do assume the format in other tests anyways so should probably fix

Reply:
Maybe we could use `moment.locale()` in `setupTests.ts` to configure this predictably? :thinking: [See this tip](https://stackoverflow.com/questions/17493309/how-do-i-change-the-language-of-moment-js).